src/game/common/bitflags.h

```diff
@@ -216,20 +217,218 @@ enum ModelConditionFlagType
+    bool No_Intersection_With(BitFlags const &set) const
+    {
+        std::bitset<bits> temp = m_bits;
+        temp &= set.m_bits;
+        return !temp.any();
+    }
 
     void Parse(INI *ini, Utf8String *string = nullptr);
-    static void Parse_INI(INI *ini, void *formal, void *store, const void *user_data);
 
-    static const char *s_bitNamesList[bits + 1];
+    void Xfer(Xfer *xfer)
+    {
+        uint8_t version = 1;
+        xfer->xferVersion(&version, 1);
+
+        if (xfer->Get_Mode() == XFER_SAVE) {
+            int count = Count();
+            xfer->xferInt(&count);
+
+            for (int i = 0; i < Size(); i++) {
+                const char *str = Get_Name_For_Bit(i);
+
+                if (str) {
+                    Utf8String s(str);
+                    xfer->xferAsciiString(&s);
+                }
+            }
+        } else if (xfer->Get_Mode() == XFER_LOAD) {
+            Clear();
+            int count;
+            xfer->xferInt(&count);
+            Utf8String str;
+
+            for (int i = 0; i < count; i++) {
+                xfer->xferAsciiString(&str);
+
+                if (!Set_Bit_By_Name(str)) {
+                    captainslog_error("invalid bit name %s", str);
+                    throw XFER_READ_ERROR;
+                }
+            }
+        } else if (xfer->Get_Mode() == XFER_CRC) {
+            xfer->xferUser(this, 4);
+        } else {
+            captainslog_error("BitFlagsXfer - Unknown xfer mode '%d'", xfer->Get_Mode());
+            throw XFER_UNKNOWN_XFER_MODE;
+        }
+    }
+
+    bool Test_For_All(BitFlags const &set)
+    {
+        captainslog_dbgassert(
+            set.Any(), "BitFlags::testForAll is always true if you ask about zero flags.  Did you mean that?");
+        std::bitset<bits> temp = m_bits;
+        temp.flip();
+        temp &= set.m_bits;
+        return !temp.any();
+    }
+
+    static int Get_Single_Bit_From_Name(const char *name);
+    bool Set_Bit_By_Name(const char *name);
+
+    const char *Get_Name_For_Bit(int bit)
+    {
+        if (Test(bit)) {
+            return s_bitNamesList[bit];
+        }
+
+        return nullptr;
+    }
+
+    void Get_Name_For_Bits(Utf8String *str)
+    {
+        if (str) {
+            for (int i = 0; i < Size(); i++) {
+                const char *c = Get_Name_For_Bit(i);
+
+                if (c) {
+                    str->Concat(c);
+                    str->Concat(",\n");
+                }
+            }
+        }
+    }
+
+    static void Parse_From_INI(INI *ini, void *formal, void *store, const void *user_data);
+    static void Parse_Single_Bit_From_INI(INI *ini, void *formal, void *store, const void *user_data);
+    static const char **Get_Bit_Names_List() { return s_bitNamesList; }
+
+    static BitFlags Set_Bit(int bit)
+    {
+        BitFlags flags;
+        flags.Set(bit, 1);
+        return flags;
+    }
+
+    static BitFlags Clear_Bit(int bit)
+    {
+        BitFlags flags;
+        flags.Set(bit, 1);
```

Comment:
think this is meant to be 0

Reply:
hmm but if it was 0 then this function is useless. What is this for? It constructs an empty bitset and sets none of its fields. Very odd..

Reply:
I realized my clone is wrong, fixed.